composite-actions/frontend-generic/upload-source-maps/scripts/upload-source-maps.js

```diff
@@ -1,157 +1,187 @@
-import fetch from "node-fetch";
-import FormData from "form-data";
-import { fileURLToPath } from "url";
+// Detect module system and import libraries dynamically
+(async () => {
+  let fs, path, fetch, FormData, fileURLToPath;
 
-const __filename = fileURLToPath(import.meta.url);
-const __dirname = path.dirname(__filename);
-
-const BUILD_DIR = process.env.BUILD_DIR;
-const PREFIX = process.env.PREFIX;
-const SERVICE_NAME = process.env.SERVICE_NAME;
-const SERVICE_VERSION = process.env.SERVICE_VERSION;
-
-const ELASTICSEARCH_URL = process.env.ELASTICSEARCH_URL;
-const ELASTICSEARCH_API_KEY = process.env.ELASTICSEARCH_API_KEY;
-
-// Helper function to get existing source maps from Elasticsearch
-async function getExistingSourceMaps() {
-  try {
-    const response = await fetch(ELASTICSEARCH_URL, {
-      method: "GET",
-      headers: {
-        Authorization: `ApiKey ${ELASTICSEARCH_API_KEY}`,
-        "Content-Type": "multipart/form-data",
-      },
-    });
-    if (!response.ok)
-      throw new Error(`Failed to fetch source maps: ${response.statusText}`);
-    return await response.json();
-  } catch (error) {
-    throw new Error(`Error fetching source maps: ${error.message}`);
-  }
-}
-
-// Function to get source maps from the build directory
-function getBuildSourceMaps() {
   try {
-    const files = fs.readdirSync(BUILD_DIR);
-
-    return files
-      .filter((file) => file.endsWith(".map"))
-      .map((file) => ({
-        mapFileName: file,
-        jsFileName: file.replace(".map", ""),
-      }));
-  } catch (error) {
-    console.error(`Error reading build directory: ${error.message}`);
-    process.exit(1);
-  }
-}
-
-// Helper function to delete a source map from Elasticsearch
-async function deleteSourceMaps(sourceMapsToDelete) {
-  const deletePromises = sourceMapsToDelete.map(async (artifactId) => {
-    try {
-      const headers = {
-        Authorization: `ApiKey ${ELASTICSEARCH_API_KEY}`,
-        "KBN-XSRF": "True",
-      };
-      const response = await fetch(`${ELASTICSEARCH_URL}/${artifactId}`, {
-        method: "DELETE",
-        headers: headers,
-      });
-      return {
-        artifactId,
-        success: response.ok,
-        statusText: response.statusText,
-      };
-    } catch (error) {
-      return { artifactId, success: false, error: error.message };
+    if (typeof require !== "undefined") {
+      // CommonJS environment
+      fs = require("fs");
+      path = require("path");
+      fetch = require("node-fetch");
+      FormData = require("form-data");
+    } else {
+      // ES Module environment
+      fs = (await import("fs")).promises;
+      path = await import("path");
+      fetch = (await import("node-fetch")).default;
+      FormData = (await import("form-data")).default;
     }
-  });
 
-  return await Promise.all(deletePromises);
-}
+    // Handle __dirname and __filename for ES Modules
+    let __dirname, __filename;
+    if (typeof __dirname === "undefined") {
```

Comment:
why did we have to do this?

Reply:
Since this action may be needed in multiple repositories that could be of any type, such as ESM or CommonJS, this condition allows us to execute the script in both environments: ESM and CommonJS.